### PR TITLE
ev-4642 Move key-cert-provisioner to the monorepo

### PR DESCRIPTION
**Pull request by rene-dekker**
Move tigera/key-cert-provisioner to the monorepo so images can be built, tagged and pushed from here.

### CODE REVIEW

File: .semaphore/push-images/key-cert-provisioner.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+version: v1.0
```
Comment (rene-dekker) [inline .semaphore/push-images/key-cert-provisioner.yml:1]: I basically copied from kube-controllers and replaced some text with key-cert-provisioner. I notice that all pipelines in calico are using ubuntu2004 now. I kept it the same.

File: .semaphore/push-images/key-cert-provisioner.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+name: Publish key-cert-provisioner images
+agent:
+  machine:
+    type: e1-standard-4
```
Comment (hjiawei) [inline .semaphore/push-images/key-cert-provisioner.yml:5]: Is `e1-standard-2` big enough for this component to build?

File: .semaphore/push-images/key-cert-provisioner.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        commands:
+        - checkout
+        # Semaphore mounts a copy-on-write FS as /var/lib/docker in order to provide a pre-loaded cache of
+        # some images. However, the cache is not useful to us and the copy-on-write FS is a big problem given
+        # how much we churn docker containers during the build.  Disable it.
+        - sudo systemctl stop docker
+        - sudo umount /var/lib/docker && sudo killall qemu-nbd || true
+        - sudo systemctl start docker
+        # Free up space on the build machine.
+        - sudo rm -rf ~/.kiex ~/.phpbrew ~/.rbenv ~/.nvm ~/.kerl ~/.sbt ~/.npm /usr/lib/jvm /opt/firefox* /opt/apache-maven* /opt/scala /usr/local/golang
```
Comment (hjiawei) [inline .semaphore/push-images/key-cert-provisioner.yml:29]:
```suggestion
        - sudo rm -rf ~/{.kerl,.kiex,.npm,.nvm,.phpbrew,.rbenv,.sbt} /opt/{apache-maven*,firefox*,scala} /usr/lib/jvm /usr/local/{aws2,golang,phantomjs*}
```

File: .semaphore/push-images/key-cert-provisioner.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      secrets:
+      - name: quay-robot-calico+semaphoreci
+      - name: docker
+      jobs:
+      - name: "key-cert-provisioner"
+        env_vars:
+        - name: DEV_REGISTRIES
+          value: quay.io/calico docker.io/calico
+        commands:
+        - if [ -z "${SEMAPHORE_GIT_PR_NUMBER}" ]; then make -C key-cert-provisioner image-all cd CONFIRM=true; fi
```
Comment (hjiawei) [inline .semaphore/push-images/key-cert-provisioner.yml:45]: Remove `image-all` and leave `cd`.

File: go.mod [hunk 5]
```diff
@@ -95,6 +95,7 @@ require (
 	k8s.io/code-generator v0.27.8
 	k8s.io/component-base v0.27.8
 	k8s.io/klog/v2 v2.90.1
+	k8s.io/kube-aggregator v0.0.0
```
Comment (hjiawei) [inline go.mod:98]: Try version `v0.27.8`?

File: key-cert-provisioner/Makefile [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+#############################################
+
+KEY_CERT_PROVISIONER_IMAGE ?=key-cert-provisioner
+TEST_SIGNER_IMAGE          ?=test-signer
+BUILD_IMAGES               ?=$(KEY_CERT_PROVISIONER_IMAGE) $(TEST_SIGNER_IMAGE)
+
+PUSH_IMAGES           ?= $(foreach registry,$(DEV_REGISTRIES),$(addprefix $(registry)/,$(BUILD_IMAGES)))
+RELEASE_IMAGES        ?= $(foreach registry,$(RELEASE_REGISTRIES),$(addprefix $(registry)/,$(BUILD_IMAGES)))
+
+GOFLAGS = -buildvcs=false
```
Comment (hjiawei) [inline key-cert-provisioner/Makefile:16]: Probably don't need this because it is part of `build_xxx` helpers.

**Review by hjiawei — APPROVED** [review]

**Review by matthewdupre — APPROVED** [review]

**Pull request merged**